cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/CassandraContext.java

```diff
@@ -96,4 +100,14 @@ protected BulkSparkConf conf()
     {
         return conf;
     }
+
+    // Startup Validation
+
+    @Override
+    public void startupValidate()
+    {
+        StartupValidator.instance().register(new SidecarValidation(sidecarClient));
+        StartupValidator.instance().register(new CassandraValidation(sidecarClient));
+        StartupValidator.instance().perform();
```

Comment:
What is the benefit of registering and performing validation immediately? 
It would be simpler to just run the 2 validations inline.

Reply:
Current implementation serves several purposes: first, we get all validation logic grouped together and easy to discover (unlike having many separate validation calls in many different places); second, we register additional validations at other places as well (see SSL/KeyStore/TrustStore validations, for example); third, we enable library users to define and register their own validations (which we ourselves use in our internal version of the library).

Reply:
Also, very easy to switch to parallel execution of validations, if ever feel need to do so in the future…

Reply:
I am OK with having it if you feel strong having it, although it feels an over-design to me.